src/mapping.ts

```diff
@@ -0,0 +1,275 @@
+import { BigInt, log } from "@graphprotocol/graph-ts";
+import {
+  Merge,
+  Transfer,
+  AlphaMassUpdate,
+  MassUpdate,
+  WhitelistUpdateCall,
+  Approval,
+  ApprovalForAll,
+  ConsecutiveTransfer,
+} from "../generated/Merge/Merge";
+import { User, NFT } from "../generated/schema";
+import { EMPTY_ADDRESS, CLASS_MULTIPLIER, NOT_WHITE_VALUE, NG_OMNIBUS } from "./utils";
+
+/* ========== HELPER FUNCTIONS ========== */
+
+/**
+ * @notice sort color of nft based on massClass
+ * @param massClass corresponding to the enum Class from schema
+ * @returns Color: string
+ */
+function checkColor(_tier: string, check: boolean): string {
+  let color: string;
+  color = "WHITE";
+  log.info("isAlpha: {}", [check.toString()]);
+
+  if (check) {
+    color = "BLACK";
+    return color;
+  } else if (_tier == "ONE") {
+    return color;
+  } else if (_tier == "TWO") {
+    color = "YELLOW";
+  } else if (_tier == "THREE") {
+    color = "BLUE";
+  } else if (_tier == "FOUR") {
+    color = "RED";
+  }
+  return color;
+}
+
+/**
+ * @notice sort tier as a string to fit schema enum reqs
+ * @param _tier
+ * @returns tier: string
+ */
+function checkMergeClass(_tier: BigInt): string {
+  let tier: string;
+  tier = "ONE";
+  if (_tier == BigInt.fromI32(1)) {
+    return tier;
+  } else if (_tier == BigInt.fromI32(2)) {
+    tier = "TWO";
+  } else if (_tier == BigInt.fromI32(3)) {
+    tier = "THREE";
+  } else if (_tier == BigInt.fromI32(4)) {
+    tier = "FOUR";
+  }
+
+  return tier;
+}
+
+// /**
+//  * @notice object type where properties represent value-based-decoded nft metadata
+//  * TODO: not sure but I am getting a compile error here. Asked DK. I need to just look into this. This will be done in next PR as I clean up the code.
+//  */
+// type Values = {
+//   massSize: BigInt;
+//   tier: string;
+//   color: string;
+// };
+
+// /**
+//  * @notice carries out decoding of value for a particular mass NFT
+//  * @returns value : Values
+//  * @dev TODO: see Object Type Values above and the error that is there. Once that is resolved then this can be used.
+//  */
+// function decodeValue(nftValue: BigInt): Values {
+//   let value: Values;
+//   let _tier: BigInt = nftValue / CLASS_MULTIPLIER;
+//   let tier = checkMergeClass(_tier);
+
+//   value = {
+//     massSize: (nftValue % CLASS_MULTIPLIER) as BigInt,
+//     tier: tier,
+//     color: checkColor(tier),
+//   };
+
+//   return value;
+// }
+
+/* ========== EVENT HANDLERS ========== */
+
+/**
+ * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
+ * @param event: Transfer
+ * @dev all NFT entity generation is handled in other functions.
+ */
+export function handleTransfer(event: Transfer): void {
+  const to = event.params.to.toHex();
+  const from = event.params.from.toHex();
+
+  let nft: NFT;
```

Comment:
you can move this after the user update happens. this is floating by itself for 20 lines

it makes the code more readable for humans

Reply:
Good point, moved it down!